gobblin-api/src/main/java/org/apache/gobblin/util/TaskEventMetadataUtils.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.util;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.runtime.api.TaskEventMetadataGenerator;
+
+public class TaskEventMetadataUtils {
+  /**
+   * Construct a {@link TaskEventMetadataGenerator} from the task state.
+   * @param taskState
+   * @return
+   */
+  public static TaskEventMetadataGenerator getTaskEventMetadataGenerator(State taskState) {
```

Comment:
Can this method be placed in other Utility class instead of creating a new one where it is the only method within the class?

Reply:
Yes, initially tried to put it in TaskUtils. Had to move it to gobblin-api module to make it available for the various classes that emit Task events.